ddl/ddl_api.go

```diff
@@ -873,24 +877,52 @@ func setDefaultAndComment(ctx context.Context, col *table.Column, options []*ast
 		return nil
 	}
 
-	for _, v := range options {
-		switch v.Tp {
+	var (
+		hasDefaultValue bool
+		hasNotNull      bool
+		setOnUpdateNow  bool
+	)
+	for _, opt := range options {
+		switch opt.Tp {
 		case ast.ColumnOptionDefaultValue:
-			err := setDefaultValue(ctx, col, v)
+			value, err := getDefaultValue(ctx, opt, col.Tp, col.Decimal)
 			if err != nil {
-				return errors.Trace(err)
+				return ErrColumnBadNull.Gen("invalid default value - %s", err)
 			}
+			col.DefaultValue = value
+			hasDefaultValue = true
 		case ast.ColumnOptionComment:
-			err := setColumnComment(ctx, col, v)
+			err := setColumnComment(ctx, col, opt)
 			if err != nil {
 				return errors.Trace(err)
 			}
+		case ast.ColumnOptionNotNull:
+			col.Flag |= mysql.NotNullFlag
+			hasNotNull = true
+		case ast.ColumnOptionNull:
+			col.Flag &= ^uint(mysql.NotNullFlag)
+		case ast.ColumnOptionOnUpdate:
+			if !expression.IsCurrentTimeExpr(opt.Expr) {
+				return ErrInvalidOnUpdate.Gen("invalid ON UPDATE for - %s", col.Name)
```

Comment:
And now we only support the type of `current_timestamp`. I will add a `TODO` here.